System.Containers.Tasks/ParseContainerProperties.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Resources;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+#nullable disable
+
+namespace System.Containers.Tasks
+{
+    public class ParseContainerProperties : Microsoft.Build.Utilities.Task
+    {
+
+        private string[] knownImageSeparators = { "-", ".", "_" };
+
+        /// <summary>
+        /// The full base image name. mcr.microsoft.com/dotnet/runtime:6.0, for example.
+        /// </summary>
+        [Required]
+        public string ContainerBaseImage { get; set; }
+
+        [Required]
+        public string ContainerImageName { get; set; }
+
+        [Required]
+        public string ContainerImageTag { get; set; }
+
+        [Output]
+        public string ParsedContainerRegistry { get; private set; }
+
+        [Output]
+        public string ParsedContainerImage { get; private set; }
+
+        [Output]
+        public string ParsedContainerTag { get; private set; }
+
+        [Output]
+        public string NewImageName { get; private set; }
+
+        [Output]
+        public string NewImageTag { get; private set; }
+
+        public override bool Execute()
```

Comment:
Style nits: I'd prefer to see the individual string manipulations here abstracted into named, individually tested methods, which should make it easier to test and also make this clearer to read.